service/src/main/java/org/apache/griffin/core/metastore/hive/HiveMetaStoreServiceJdbcImpl.java

```diff
@@ -0,0 +1,344 @@
+    @Cacheable(unless = "#result==null")
+    public Map<String, List<String>> getAllTableNames() {
+//        // Not recommend this method because this method will generate one query for every database
+//        // If there has a lots of databases in Hive, this method will lead to Griffin crash
+//        Map<String, List<String>> res = new HashMap<>();
+//
+//        for (String dbName : getAllDatabases()) {
+//            List<String> list = (List<String>) queryHiveString(SHOW_TABLES_IN + dbName);
+//            res.put(dbName, list);
+//        }
+//
+//        return res;
+        Map<String, List<String>> res = new HashMap<>();
+
+        List<String> list = new ArrayList<>(Arrays.asList("merch_data", "merch_summary_ext_v3"));
+        res.put("default", list);
+        return res;
+    }
+
+    @Override
+    public List<Table> getAllTable(String db) {
+        return null;
+    }
+
+    @Override
+    public Map<String, List<Table>> getAllTable() {
+        return null;
+    }
+
+    @Override
+    @Cacheable(unless = "#result==null")
+    public Table getTable(String dbName, String tableName) {
+        Table result = new Table();
+        result.setDbName(dbName);
+        result.setTableName(tableName);
+
+        String sql = SHOW_CREATE_TABLE + dbName + "." + tableName;
+        Statement stmt = null;
+        ResultSet rs = null;
+        StringBuilder sb = new StringBuilder();
+
+        try {
+            Class.forName(hiveClassName);
+            if (conn == null) {
+                conn = DriverManager.getConnection(hiveUrl);
+            }
+            LOGGER.info("got connection");
+
+            stmt = conn.createStatement();
+            rs = stmt.executeQuery(sql);
+            while (rs.next()) {
+                String s = rs.getString(1);
+                sb.append(s);
+            }
+            String location = getLocation(sb.toString());
+            List<FieldSchema> cols = getColums(sb.toString());
+            StorageDescriptor sd = new StorageDescriptor();
+            sd.setLocation(location);
+            sd.setCols(cols);
+            result.setSd(sd);
+        } catch (Exception e) {
+            LOGGER.error("Query Hive Table metadata has error. {}", e.getMessage());
+        } finally {
+            closeConnection(stmt, rs);
+        }
+        return result;
+    }
+
+    @Scheduled(fixedRateString =
+            "${cache.evict.hive.fixedRate.in.milliseconds}")
+    @CacheEvict(
+            cacheNames = "jdbcHive",
+            allEntries = true,
+            beforeInvocation = true)
+    public void evictHiveCache() {
+        LOGGER.info("Evict hive cache");
+    }
+
+    /**
+     * Query Hive for Show tables or show databases, which will return List of String
+     * @param sql sql string
+     * @return
+     */
+    private Iterable<String> queryHiveString(String sql) {
+        List<String> res = new ArrayList<>();
+        Statement stmt = null;
+        ResultSet rs = null;
+
+        try {
```

Comment:
would you like to consider refactoring db access code snippet into a common method

Reply:
I know it's not an elegant solution, but is a working solution. I have plan to change the db access code in future.